src/api/posts.ts

```diff
@@ -0,0 +1,25 @@
+import { Post } from '../react-app-env';
+
+export const BASE_URL = 'https://mate.academy/students-api/';
+
+export async function getPosts(userId: string): Promise<Post[]> {
+  const response = await fetch(`${BASE_URL}/posts?userId=${userId}`);
+
+  if (!response.ok) {
+    throw new Error(`${response.status} - ${response.statusText}`);
+  }
+
+  return response.json();
+}
+
+export async function getPostbyId(postId:number) {
```

Comment:
Requirement was to create ```getPostDetails(postId)``` function. I guess that `getPostById` is also ok but personally i wouldn't use that becasue as you can imagine you can fetch post by Id but maybe you have and API which can return different information about post. ```getPostDetails(postId)```clearly indentifies what you get.